### PR TITLE
Unit testing

### PR DESCRIPTION
**Pull request by anjaradenovic**

### CODE REVIEW

File: src/test/java/com/endava/algoritm/SolutionTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    @Test
+    void testAllZeroes(){
+        int expected = 0;
+        int[] array = {0, 0, 0};
+        assertEquals(expected, solution.solution(array));
+    }
+
+    @Test
+    void testOneIsZero(){
```
Comment (oskarv) [inline src/test/java/com/endava/algoritm/SolutionTest.java]: missing indentation after `()`

File: src/test/java/com/endava/algoritm/SolutionTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        int expected = 0;
+        int[] array = {0, 2, 3};
+        assertEquals(expected, solution.solution(array));
+    }
+
+    @Test
+    void testAllOnes(){
```
Comment (oskarv) [inline src/test/java/com/endava/algoritm/SolutionTest.java]: missing indentation after ()

File: src/test/java/com/endava/algoritm/SolutionTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        int expected = 1;
+        int[] array = {1, 1, 1};
+        assertEquals(expected, solution.solution(array));
+    }
+
+    @Test
+    void testFourNumbers(){
```
Comment (oskarv) [inline src/test/java/com/endava/algoritm/SolutionTest.java]: missing indentation after ()

File: src/test/java/com/endava/algoritm/SolutionTest.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        int expected = 24;
+        int[] array = {4, 3, 2, 2};
+        assertEquals(expected, solution.solution(array));
+    }
+
+    @Test
+    void testFourNumbersTwoNegative(){
```
Comment (oskarv) [inline src/test/java/com/endava/algoritm/SolutionTest.java]: missing indentation after ()

File: src/test/java/com/endava/algoritm/SolutionTest.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        int expected = 40;
+        int[] array = {-5, -4, 2, 2};
+        assertEquals(expected, solution.solution(array));
+    }
+
+    @Test
+    void testFourNumbersTwoBigNegativeOneBigPossitive(){
```
Comment (oskarv) [inline src/test/java/com/endava/algoritm/SolutionTest.java]: missing indentation after ()

File: src/main/java/com/endava/algoritm/Solution.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
+package com.endava.algoritm;
+
+public class Solution {
+
+    public static int solution(int[] A) {
+        int minMin = Integer.MAX_VALUE;
+        int min = Integer.MAX_VALUE;
+        int max = Integer.MIN_VALUE;
+        int maxMax = Integer.MIN_VALUE;
+        int maxMaxMax = Integer.MIN_VALUE;
```
Comment (oskarv) [inline src/main/java/com/endava/algoritm/Solution.java]: what would be better naming convention here?
maxMax, maxMaxMax, minMin are not descriptive and someone reading the code would have no idea what is their purpose.

**Review by oskarv — CHANGES_REQUESTED** [review]

File: src/main/java/com/endava/algoritm/Solution.java (hunk 6 above)
Comment (anjaradenovic) [reply src/main/java/com/endava/algoritm/Solution.java]: I do not know how to name it :( First it was min1, min2... then I realized it is confusing for me. Any suggestions are appreciated :)

File: src/test/java/com/endava/algoritm/SolutionTest.java (hunk 1 above)
Comment (anjaradenovic) [reply src/test/java/com/endava/algoritm/SolutionTest.java]: tnx

File: src/test/java/com/endava/algoritm/SolutionTest.java (hunk 3 above)
Comment (anjaradenovic) [reply src/test/java/com/endava/algoritm/SolutionTest.java]: tnx

File: src/main/java/com/endava/algoritm/Solution.java (hunk 6 above)
Comment (oskarv) [reply src/main/java/com/endava/algoritm/Solution.java]: Can you describe in a sentence or two, what the use for maxMax and what is the use for maxMaxMax?

Comment (anjaradenovic) [reply src/main/java/com/endava/algoritm/Solution.java]: The more **max**'s there are in the name, number is bigger.
The more **min**'s there are in the name, number is smaller. 
:)

Maybe to add comment like this:
```
/*
Order from smaller to bigger:
minMin <= min <= max <= maxMax <= maxMaxMax
*/
```

Comment (oskarv) [reply src/main/java/com/endava/algoritm/Solution.java]: I don't like comments, but that's just my code style. Comment is valid option here.
If i wrote the code i would name them: max, secondMax ...
secondMax clearly tells me that that's second biggest number;

Comment (anjaradenovic) [reply src/main/java/com/endava/algoritm/Solution.java]: Make sense. Code changed.

**Review by oskarv — APPROVED** [review]
👍

**Pull request merged**